### PR TITLE
Remove S3 dependency and enable local integ testing  

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@dblock.org>

### Description

- Replaces dependency on S3 by downloading manifests and artifacts from public URLs.
- Enables running `./test.sh integ-test .` that uses a local build and distribution to execute tests.
 
Note that tests for 1.2.0 will continue failing because of DistributionDownloader, but all the dependencies are properly downloaded and installed from local or remote.

```
Cause 1: org.gradle.internal.resolve.ModuleVersionResolveException: Could not resolve opensearch-distribution:opensearch:1.2.0.
Caused by: org.gradle.api.resources.ResourceException: Could not get resource 'https://artifacts.opensearch.org/releases/core/opensearch/1.2.0/opensearch-min-1.2.0-linux-x64.tar.gz'.
```

### Issues Resolved

- Closes https://github.com/opensearch-project/opensearch-build/issues/629
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/run_perf_test.py [hunk 1]
```diff
@@ -11,23 +11,6 @@
 from test_workflow.perf_test.perf_test_cluster import PerfTestCluster
 from test_workflow.perf_test.perf_test_suite import PerfTestSuite
 
-"""
```
Comment (dblock) [inline src/run_perf_test.py:14]: This made this implementation untestable and untested. Whenever you require the file it would run through this code.

Comment (owaiskazi19) [reply src/run_perf_test.py:14]: Thanks for catching that.

File: Pipfile [hunk 2]
```diff
@@ -13,17 +13,14 @@ isort = "~=5.9"
 flake8 = "~=3.9"
 mypy = "~=0.9"
 pytest = "*"
-boto3 = "1.18.31"
```
Comment (peternied) [inline Pipfile:16]: yay!

**Review by peternied — APPROVED** [review]
Nicely done

**Review by owaiskazi19 — APPROVED** [review]
LGTM!

**Pull request merged**